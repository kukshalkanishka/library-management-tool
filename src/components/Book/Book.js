import React from "react";

const Book = ({name, details, toggleTodoState, deleteTodo}) => {
  return (
    <div className="todoContainer">
      <div className={`todoText ${details.number ? "" : "done"}`}>{name}</div>
      <button
        className="markAsDone"
        onClick={deleteTodo.bind(null, {name})}
      >{`\u{1F5D1}`}</button>
    </div>
  );
};

export default Book;
