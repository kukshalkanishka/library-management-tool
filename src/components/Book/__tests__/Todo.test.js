import React from "react";
import { shallow, configure } from "enzyme";
import Book from "../Todo";

import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("Todo", () => {
  it("should render properly", () => {
    const todo = { todo: "dummyTodo", date: "1234" };
    const toggleTodoState = jest.fn();
    const deleteTodo = jest.fn();
    const wrapper = shallow(
      <Book
        todo={todo}
        toggleTodoState={toggleTodoState}
        deleteTodo={deleteTodo}
      />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
