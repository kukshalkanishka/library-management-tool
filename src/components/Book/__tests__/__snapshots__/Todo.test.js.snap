// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo should render properly 1`] = `
"<div className=\\"todoContainer\\">
  <input type=\\"checkbox\\" onClick={[Function: bound mockConstructor]} checked={false} />
  <div className=\\"todoText \\">
    dummyTodo
  </div>
  <button className=\\"markAsDone\\" onClick={[Function: bound mockConstructor]}>
    ðŸ—‘
  </button>
</div>"
`;
